cmd/controller/controller.go

```diff
@@ -270,12 +270,11 @@ func isNotFound(err error) bool {
 }
 
 func resolveURL(base, ref string) (string, error) {
-	refURL, err := url.Parse(ref)
+	base = strings.TrimSpace(base)
+	ref = strings.TrimSpace(ref)
+	refURL, err := url.ParseRequestURI(ref)
 	if err != nil {
-		baseURL, err := url.Parse(base)
-		if err != nil {
-			return "", err
-		}
+		baseURL, _ := url.Parse(base)
```

Comment:
Is there a reason you're discarding the error here?  It will be very confusing for someone when the base URL is invalid, fails to parse here, but then continues and triggers a generic segfault crash a few lines below because `baseURL` was nil